src/GUI/UI/UIDialogue.cpp

```diff
@@ -644,14 +642,13 @@ void OnSelectNPCDialogueOption(DIALOGUE_TYPE option) {
                 pParty->TakeGold(pNPCStats->pProfessions[speakingNPC->profession].uHirePrice);
             }
             speakingNPC->uFlags |= 0x80u;
-            if (pParty->pHirelings[0].pName) {
-                memcpy(&pParty->pHirelings[1], speakingNPC, sizeof(pParty->pHirelings[1]));
-                v13 = pParty->pHireling2Name;
+            if (!pParty->pHirelings[0].pName.empty()) {
+                pParty->pHirelings[1] = *speakingNPC;
+                pParty->pHireling2Name = speakingNPC->pName;
             } else {
-                memcpy(&pParty->pHirelings[0], speakingNPC, sizeof(pParty->pHirelings[0]));
-                v13 = pParty->pHireling1Name;
+                pParty->pHirelings[0] = *speakingNPC;
```

Comment:
There is pParty->pHirelings[x].pName corruption, the same in the block above.

Reply:
Eh, I'm not seeing it. Can you provide more details?

Reply:
Can't really see it. Can you provide more details?

Reply:
Hire any npc then try to talk with this npc, his name will be garbaged and more information dialog will cause a crash